lib/components/SelectionControl/ToggleSwitch/toggle-switch.scss

```diff
@@ -0,0 +1,89 @@
+@import "@quill/color.scss";
+@import "@quill/color.opacity.scss";
+@import "@quill/others.scss";
+@import "@quill/border.scss";
+@import "@quill/motion.scss";
+@import "@quill/size.scss";
+@import "@quill/spacing.scss";
+@import "@quill/color.semantic.scss";
+
+:root {
+    .theme--light {
+        --switch-background-color: var(--core-color-opacity-black-200);
+        --switch-active-color: var(--core-color-opacity-black-400);
+        --switch-enabled: var(--core-color-opacity-green-600);
+        --switch-active-enabled-color: var(--core-color-solid-green-900);
+        --switch-hover-enabled-color: var(--core-color-opacity-green-800);
+    }
+    .theme--dark {
+        --switch-background-color: var(--core-color-opacity-white-400);
+        --switch-active-color: var(--core-color-opacity-white-200);
+        --switch-enabled: var(--core-color-opacity-green-800);
+        --switch-active-enabled-color: var(--core-color-opacity-green-400);
+        --switch-hover-enabled-color: var(--core-color-opacity-green-600);
+    }
+}
+
+.toggle-switch {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    position: relative;
+    user-select: none;
+    width: var(--core-size-2400);
+    height: var(--core-size-1400);
+    border-width: var(--core-borderRadius-100);
+    &__disabled .toggle-switch__slider {
+        pointer-events: none;
+        background: var(--semantic-color-typography-bg);
+    }
+    &__slider {
+        display: flex;
+        align-items: center;
+        justify-content: flex-start;
+        width: var(--core-size-2200);
+        height: var(--size-generic-sm);
+        background-color: var(--switch-background-color);
+        border-radius: var(--semantic-borderRadius-lg);
+        transition: background-color var(--core-motion-duration-300) ease;
+        cursor: pointer;
+        position: relative;
+        &:hover:not(.__disabled) {
+            background-color: var(--semantic-color-typography-disabled);
+        }
+
+        &:active:not(.__disabled) {
+            background-color: var(--switch-active-color);
+        }
+
+        &.__enabled {
+            background-color: var(--switch-enabled);
+            &:hover:not(.__disabled) {
+                background-color: var(--switch-hover-enabled-color);
+            }
+
+            &:active:not(.__disabled) {
+                background-color: var(--switch-active-enabled-color);
+            }
+        }
+    }
+    &__knob {
+        position: absolute;
+        width: var(--core-size-1000);
+        height: var(--core-size-1000);
+        border-radius: var(--core-borderRadius-2500);
+        background-color: var(--core-color-solid-slate-50);
+        transition: left var(--core-motion-duration-300) ease;
+        top: 50%;
+        transform: translateY(-50%);
+        left: var(--core-spacing-100);
+
+        &.__enabled {
+            left: calc(100% - 22px);
```

Comment:
22px is half of the width of .`toggle-switch__slider` (like `calc(var(--core-size-2200) / 2)`)?